connector/src/main/scala/com/microsoft/kusto/spark/utils/ContainerProvider.scala

```diff
@@ -21,13 +22,12 @@ class ContainerProvider(val client: ExtendedKustoClient, val clusterAlias: Strin
   private var lastRefresh: Instant = Instant.now(Clock.systemUTC())
   private val className = this.getClass.getSimpleName
   private val maxCommandsRetryAttempts = 8
-  private val retryConfig = buildRetryConfig
-
-  private def buildRetryConfig = {
-    val retryException: Predicate[Throwable] = (e: Throwable) =>
-      (e.isInstanceOf[IngestionServiceException] && !e.asInstanceOf[KustoDataExceptionBase].isPermanent) ||
-        (e.isInstanceOf[DataServiceException] && ExceptionUtils.getRootCause(e).isInstanceOf[HttpHostConnectException])
+  private val retryConfig = buildRetryConfig((e: Throwable) =>
+    (e.isInstanceOf[IngestionServiceException] && !e.asInstanceOf[KustoDataExceptionBase].isPermanent) ||
+      (e.isInstanceOf[DataServiceException] && ExceptionUtils.getRootCause(e).isInstanceOf[HttpHostConnectException]))
+  private val retryConfigIngestionRefresh = buildRetryConfig(e => e.isInstanceOf[RuntimeException])
```

Comment:
After we change it to a different type of exception change it here as well and add here all the SDK exceptions as well

Reply:
Still here RuntimeException  - need to be only the IngestionServiceException, IngestionClientException